src/modules/site-settings/edit-site-details.tsx

```diff
@@ -54,7 +54,23 @@ export default function EditSiteDetails( { currentWpVersion, onSave }: EditSiteD
 		selectedSite?.customDomain ?? null
 	);
 	const [ customDomainError, setCustomDomainError ] = useState( '' );
+	const [ existingDomainNames, setExistingDomainNames ] = useState< string[] >( [] );
 	const [ enableHttps, setEnableHttps ] = useState( false );
+
+	useEffect( () => {
+		getIpcApi()
+			.getAllCustomDomains()
+			.then( ( domains ) => {
+				const domainsWithoutSelectedSite = domains.filter(
+					( domain ) => domain !== selectedSite?.customDomain
+				);
+				setExistingDomainNames( domainsWithoutSelectedSite );
+			} )
+			.catch( () => {
+				// Do nothing
+			} );
+	}, [ selectedSite?.customDomain ] );
```

Comment:
It would be cool to extract this code into a `useAvailableCustomDomains()` hook to improve readability.